Complete homework

Wathe6 opened the pull request:
Готовое задание на ветке для проверки.

Wathe5 committed:
Complete homework

Готовое задание на ветке для проверки.

YlabCourse commented on Homework1_Habits/.gradle/7.5.1/checksums/checksums.lock:
Лишние файлы в PR, используй .gitignore файл для того чтобы файлы .gradle и .idea не добавлялись в проект

YlabCourse commented on Homework1_Habits/src/main/java/io/envoi/Main.java:
```diff
@@ -0,0 +1,141 @@
+import java.time.format.DateTimeParseException;
+import java.util.*;
+
+public class Main
```
Main - это класс в котором у тебя расположена точка входа в приложение и он должен содержать только запуск твоей программы. То есть всю логику с вводом выводом регистрацией и прочим нужно вынести в другие классы. Не стоит нарушать принцип Single-responsibility из SOLID.

YlabCourse commented on Homework1_Habits/src/main/java/io/envoi/Main.java:
```diff
@@ -0,0 +1,141 @@
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.*;
```
Не используй Wildcards, лучше явно прописывай импорты.

YlabCourse commented on Homework1_Habits/src/main/java/io/envoi/Main.java:
```diff
@@ -0,0 +1,141 @@
+                {
+                    return input;
+                }
+                System.out.println("Число должно быть между " + min + " и " + max + ".");
```
Лучше использовать форматирования например класс String.format() или System.out.printf() а не делать конкатенацию прямо в методе вывода в консоль

YlabCourse commented on Homework1_Habits/src/main/java/io/envoi/auth/Authentication.java:
```diff
@@ -0,0 +1,344 @@
+        System.out.println("Имя успешно изменено!");
+        return true;
+    }
+
```
Где-то пробелы есть, а где-то нету. Лучше везде между методами ставить пробелы

YlabCourse commented on Homework1_Habits/src/main/java/io/envoi/menu/HabitsMenu.java:
```diff
@@ -0,0 +1,186 @@
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.envoi.Main.*;
```
тоже Wildcards, пишу здесь замечание чтобы убрал во всём проекте

YlabCourse commented on Homework1_Habits/src/main/java/io/envoi/model/Account.java:
```diff
@@ -0,0 +1,80 @@
+import java.util.Objects;
+
+public class Account implements Serializable
+{
```
В java принято располагать скобочку на одной строке с классом

YlabCourse commented on Homework1_Habits/src/main/java/io/envoi/model/Account.java:
```diff
@@ -0,0 +1,80 @@
+import java.io.Serializable;
+import java.util.Objects;
+
+public class Account implements Serializable
```
По требованиям первого задания данные должны храниться в коллекциях java, сериализации не должно быть

YlabCourse commented on Homework1_Habits/src/main/java/io/envoi/model/Account.java:
```diff
@@ -0,0 +1,80 @@
+        setRole(role);
+    }
+
+    public String getEmail()
```
используй библиотеку lombok для генерации геттеров сеттеров и прочего

YlabCourse commented on Homework1_Habits/build/reports/tests/test/packages/model.html:
```diff
@@ -0,0 +1,123 @@
+<td>
+<div class="infoBox" id="tests">
+<div class="counter">25</div>
+<p>tests</p>
```
Лишние файлы, почитай насчёт того что должно быть в репозитории, а что не нужно добавлять. Добавь все лишние файлы в .gitignore файл

YlabCourse commented on Homework1_Habits/src/main/java/io/envoi/persistence/AccountPersistence.java:
```diff
@@ -0,0 +1,48 @@
+        }
+    }
+
+    @SuppressWarnings("unchecked")
```
Данные должны храниться в коллекциях (в памяти).

YlabCourse commented on Homework1_Habits/src/main/java/io/envoi/persistence/UsersPersistence.java:
```diff
@@ -0,0 +1,48 @@
+            System.err.println("Ошибка при сохранении профилей: " + e.getMessage());
+        }
+    }
+
```
Данные должны храниться в коллекциях (в памяти).

YlabCourse commented on Homework1_Habits/src/main/java/io/envoi/service/AccountsService.java:
```diff
@@ -0,0 +1,90 @@
+import java.util.HashMap;
+import java.util.Map;
+
+public class AccountsService
```
Добавь абстракцию в виде интерфейсов для сервисов (dependency inversion principle SOLID)

YlabCourse commented on Homework1_Habits/src/main/java/io/envoi/service/UsersService.java:
```diff
@@ -0,0 +1,65 @@
+
+public class UsersService
+{
+    private Map<String, User> users;
```
Лучше вынести хранение данных в репозиторный слой и вызывать репозиторий из сервиса (сделай зависимость от интерфесов а не от реализаций)

YlabCourse commented on Homework1_Habits/src/test/java/model/AccountTest.java:
```diff
@@ -0,0 +1,118 @@
+        account = new Account(EMAIL, PASSWORD, ROLE);
+    }
+
+    @Test
```
добавь аннотацию @DisplayName чтобы наглядно видеть название тестов при запуске

YlabCourse commented on Homework1_Habits/src/test/java/model/AccountTest.java:
```diff
@@ -0,0 +1,118 @@
+    void testEquals_identicalAccount()
+    {
+        Account identicalAccount = new Account(EMAIL, PASSWORD, ROLE);
+        assertEquals(account, identicalAccount);
```
используй лучше assertj для более гибкой проверки в тестах

YlabCourse commented on Homework1_Habits/src/main/java/io/envoi/auth/Authentication.java:
```diff
@@ -0,0 +1,344 @@
+import static io.envoi.Main.inputInt;
+import static io.envoi.Main.inputLine;
+
+public class Authentication
```
Отсутствует java doc в проекте

YlabCourse reviewed (commented):
В целом логика, не плохая, но форматирование кода страдает. Придерживайся конвенций форматирования для java кода, если пишешь на java =) Создай файл .gitignore и добавь все лишние файлы в него, чтобы не загрязнять репозиторий лишним. Добавь java doc. Добавь зависимость компонентов на абстракцию (интерфейсы) а не на реализацию. Main класс сделай ответственным только за запуск приложения и делегации прочей логики в другие классы.

Wathe5 committed:
Correction

Added javadoc descriptions for classes.
Added DisplayNames in tests.
Added BasicService for AccountsService and UsersService.
Added .gitignore
Main was transformed into MainMenu.
println was changed for format.
